### PR TITLE
only display 'Engine Version' when it's not empty

### PR DESCRIPTION
**Pull request by vieux**
Please keep in mid that the cli doesn't always talk to an engine, sometimes it's swarm :D

### CODE REVIEW

**Comment by vieux** [comment]
ping @tiborvass

File: api/client/info.go [hunk 1]
```diff
@@ -35,7 +35,9 @@ func (cli *DockerCli) CmdInfo(args ...string) error {
 
 	fmt.Fprintf(cli.out, "Containers: %d\n", info.Containers)
 	fmt.Fprintf(cli.out, "Images: %d\n", info.Images)
-	fmt.Fprintf(cli.out, "Engine Version: %s\n", info.ServerVersion)
+	if info.ServerVersion != "" {
+		fmt.Fprintf(cli.out, "Engine Version: %s\n", info.ServerVersion)
+	}
```
Comment (thaJeztah) [inline api/client/info.go]: Should this use `ioutils.FprintfIfNotEmpty`? (I just spotted that the line below)

Comment (duglin) [inline api/client/info.go]: +1

**Comment by cpuguy83** [comment]
What about `Daemon Version: <daemon name>/<daemon version>`, which is similar to what you currently do for the server portion of `docker version`.. which I think is the intent of adding this to info.

File: api/client/info.go (hunk 1 above)
Comment (vieux) [inline api/client/info.go]: I'll update

**Comment by vieux** [comment]
@cpuguy83 I would be fine with daemon.

@tiborvass @jfrazelle @icecrime ?

**Comment by vieux** [comment]
@cpuguy83 actually `docker version` uses `server` not `daemon`

**Comment by tiborvass** [comment]
lol, such chauvinism these days :D

@vieux I agree with @cpuguy83 that we should be consistent with `docker version`. Having said that, I'm not sure if a *version* has to have a name in it. But to unblock this I'm LGTM-ing we can always bikeshed another time.

**Comment by thaJeztah** [comment]
LGTM, thanks @vieux

**Comment by tiborvass** [comment]
Just to make it clear I agree with `Server Version`

**Comment by vieux** [comment]
@tianon @thaJeztah I switched to `Server Version` PTAL

**Comment by cpuguy83** [comment]
LGTM

Also my point was really it would be good to get the swarm info in this output rather than have nothing, hence having the daemon name (engine or swarm), that's all.

**Comment by tiborvass** [comment]
@vieux will this output `Server Version: swarm/v1.0` ?

**Comment by vieux** [comment]
@tiborvass yes, like in version.

**Comment by tiborvass** [comment]
Cool

**Comment by thaJeztah** [comment]
Hm, this now needs doc-changes https://github.com/docker/docker/search?utf8=✓&q=Engine+version

**Comment by vieux** [comment]
@thaJeztah I updated the doc

**Comment by thaJeztah** [comment]
Thanks! LGTM

**Comment by thaJeztah** [comment]
Putting it back to "merge", don't think this needs another docs review :innocent:

**Pull request merged**